invitev2.go

```diff
@@ -0,0 +1,125 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/tidwall/gjson"
+)
+
+// InviteV2Request and InviteV2StrippedState are defined in
+// https://matrix.org/docs/spec/server_server/r0.1.3#put-matrix-federation-v2-invite-roomid-eventid
+
+func NewInviteV2Request(event *HeaderedEvent, state []InviteV2StrippedState) (
+	request InviteV2Request, err error,
+) {
+	if event.RoomVersion == "" {
+		err = errors.New("gomatrixserverlib: malformed headered event")
+		return
+	}
+	request.fields.inviteV2RequestHeaders = inviteV2RequestHeaders{
+		RoomVersion:     event.RoomVersion,
+		InviteRoomState: state,
+	}
+	request.fields.Event = event.Unwrap()
+	return
+}
+
+type inviteV2RequestHeaders struct {
+	RoomVersion     RoomVersion             `json:"room_version"`
+	InviteRoomState []InviteV2StrippedState `json:"invite_stripped_state"`
+}
+
+// InviteV2Request is used in a /_matrix/federation/v2/invite request.
```

Comment:
If it's the body, just say it is. If it's a mix, say so too.

Reply:
Not sure I understand what you mean here?

Reply:
This struct is used in a request. Where in the request? Query params? Body? Headers?

Reply:
Ah OK, in the body it is - fixed.